Added Testimonial Section Desktop + Mobile

mike10911 opened the pull request:
Added Testimonial Section

Fixes #34 

Changelist:
- `TestimonialsSection.tsx `- Entire TestimonialsSection called in the index
- `TestimonialsSection.styles.ts` - Styles for `TestimonialsSection.tsx`
- `Testimonials.tsx` - Component for the group of 3 visible testimonials
- `Testimonials.styles.ts` - Styles for `Testimonials.tsx`
- `TestimonialCard.tsx `- Component for the single testimonial in the center
- `LeftOrRightTestimonialCard.tsx` - Component for two side testimonials.
- `TestimonialCard.styles.ts` - Styles for both `TestimonialCard.tsx` & `LeftOrRightTestimonialCard.tsx`
- Added new images
- Added`getLeftOrRightTestimonial()` in `utils.is`
- Added `TEXT_GREY` in `colors.ts`

Notes:

- Still need to figure out the cacti getting bigger when selected.

Tested:

- Local Host in multiple browsers

Screenshots & Screencasts:
<img width="1800" alt="Screen Shot 2022-10-18 at 6 49 19 PM" src="https://user-images.githubusercontent.com/103906123/196559748-133dc091-f74f-40f6-8295-d241f9fda629.png">

mike10911 committed:
testimonial section

mike10911 committed:
build fix

mike10911 committed:
build fix

jzhzhang66 commented on main-site/components/testimonials-section/testimonials/Testimonials.tsx:
```diff
@@ -0,0 +1,149 @@
+  const [currentIndex, setCurrentIndex] = useState<number>(0);
+  const goToTestimonial = (index: number): void => {
+    setCurrentIndex(index);
+    console.log(index);
```
remove all console logs (there's several other ones)

jzhzhang66 commented on main-site/components/testimonials-section/testimonials/Testimonials.tsx:
```diff
@@ -0,0 +1,149 @@
+                'left',
+                testimonialSectionData,
+                currentIndex
+              ).quote
```
Can you create a const on like line 39 leftTestimonial and set it equal to    getLeftOrRightTestimonial(
                'left',
                testimonialSectionData,
                currentIndex
              ) ?
 
There's no need to call it 5 times if we can just call it once 
Also, you can just pass in one prop testimonial={leftTestimonial} and then access the parameters in your LeftOrRightTestimonialCard since they all come from the same object anyways there's no need to pass them in separately here.

jzhzhang66 commented on main-site/components/testimonials-section/testimonials/Testimonials.tsx:
```diff
@@ -0,0 +1,149 @@
+                testimonialSectionData,
+                currentIndex
+              ).quote
+            }
```
Same here as what I mentioned above

jzhzhang66 commented on main-site/lib/utils.ts:
```diff
@@ -21,3 +21,29 @@ export function getLeftOrRight(
+  currentIndex: number
+): TestimonialData {
+  const testimonialDatalength = testimonialData.length;
+  console.log(currentIndex);
```
remove console log

jzhzhang66 reviewed (changes requested):
Mike this looks great!!!! Awesome job figuring out the logic and everything you are a react king!! Just a few comments - 

- For some reason the right container is going off the screen if you scroll to the right so investigate to see whats going on with that. I think I mightve had the same issue with the about section i can't remember what fixed it but maybe take a look in the about section styling to see if there's anything that might fix this.
<img width="1112" alt="Screen Shot 2022-10-18 at 8 42 51 PM" src="https://user-images.githubusercontent.com/59738880/196571050-9b6561d4-7a9e-4fc7-a6fa-b3d293a75330.png">
- The description text is really small right now - I know it's because the testimonials are longer than the ones in the design but we should try to figure out a workaround. Each testimonial in the design does look bigger than the website though - see video. Let's make each testimonial bigger and then increase the font size as much as we can  

https://user-images.githubusercontent.com/59738880/196571640-a08789b4-6b2d-4de7-aebb-07f1b16e79ee.mov
- between tablet and tabletLg, there isn't any space between the different testimonials. Can you move the left one more left and the right one more right on these screensizes to create space? 
<img width="1133" alt="Screen Shot 2022-10-18 at 8 51 57 PM" src="https://user-images.githubusercontent.com/59738880/196571967-8cfc0d7b-9ead-42df-8ec5-155166c5ed42.png">

deanframe commented on main-site/components/testimonials-section/testimonial-card/TestimonialCard.tsx:
```diff
@@ -0,0 +1,54 @@
+        <StyledTestimonialCardBox>
+          <StyledTestimonialCardImage src={getImage(author)} />
+          <StyledTestimonialCardAuthor>
+            <p>{author}</p>
```
dumb question because i've been in the app portal for too long: is there a custom styled `<P>` tag that we should be using instead?

mike10911 replied:
Yes you are right! I used a custom tag in `StyledTestimonialCardAuthor` and forgot to remove the  `<p>`. Nice catch!

deanframe commented on main-site/lib/data.ts:
```diff
@@ -96,3 +96,35 @@ export const FaqSectionData: FaqData[] = [
+  {
+    id: 0,
+    author: 'Karyna Yen',
+    year: 'Northeastern 2025',
```
maybe say c/o 2025 or '25 here instead?

mike10911 replied:
I think '25 looks better, what do you think? @deanframe 

<img width="432" alt="Screen Shot 2022-10-21 at 12 56 13 PM" src="https://user-images.githubusercontent.com/103906123/197249165-88ffab43-7bae-44ab-a1da-4a63e7420b3a.png">

jzhzhang66 replied:
^^ i agree '25 looks better

deanframe replied:
+1 for '25

deanframe commented on main-site/lib/data.ts:
```diff
@@ -96,3 +96,35 @@ export const FaqSectionData: FaqData[] = [
+    author: 'Karyna Yen',
+    year: 'Northeastern 2025',
+    quote:
+      '"I had a ton of fun at HackBeanpot! We made a Chrome extension that censored inappropriate language on the internet. My favorite part was went one of my teammates got an animated duck to quack when clicking on the extension icon!"'
```
fix typo - "was went"

deanframe reviewed (commented):
mostly looks great, some small stuff in addition to what judy said

mike10911 commented:
QUICK QUESTION!
In the design, the quotes have different fonts depending on the length. Should I have the font change to fit the card or should I just make the font as big possible? @deanframe @jzhzhang66

jzhzhang66 commented:
@mike10911 just make it the biggest font possible!!!

mike10911 committed:
remove console logs

mike10911 committed:
NE class update

mike10911 committed:
cleaned up LeftOrRightTestimonial props

mike10911 committed:
fixed karyna quote typo

mike10911 committed:
increased quote font size

mike10911 committed:
removed p tag from StyledTestimonialCardAuthor

mike10911 committed:
right card fix

mike10911 commented:
> Mike this looks great!!!! Awesome job figuring out the logic and everything you are a react king!! Just a few comments -
> 
> * For some reason the right container is going off the screen if you scroll to the right so investigate to see whats going on with that. I think I mightve had the same issue with the about section i can't remember what fixed it but maybe take a look in the about section styling to see if there's anything that might fix this.
> 
> <img alt="Screen Shot 2022-10-18 at 8 42 51 PM" width="1112" src="https://user-images.githubusercontent.com/59738880/196571050-9b6561d4-7a9e-4fc7-a6fa-b3d293a75330.png">
> 
> - The description text is really small right now - I know it's because the testimonials are longer than the ones in the design but we should try to figure out a workaround. Each testimonial in the design does look bigger than the website though - see video. Let's make each testimonial bigger and then increase the font size as much as we can
>  Screen.Recording.2022-10-18.at.8.48.20.PM.mov 
> * between tablet and tabletLg, there isn't any space between the different testimonials. Can you move the left one more left and the right one more right on these screensizes to create space?
> 
> <img alt="Screen Shot 2022-10-18 at 8 51 57 PM" width="1133" src="https://user-images.githubusercontent.com/59738880/196571967-8cfc0d7b-9ead-42df-8ec5-155166c5ed42.png">

mike10911 commented:
Hey @jzhzhang66 

Currently trying to fix the overflow issue, making the position of the section container got rid of most of it but there is still a sliver of overflow left. Is this coming from the past photos component or still testimonials?

<img width="70" alt="Screen Shot 2022-10-22 at 12 35 40 AM" src="https://user-images.githubusercontent.com/103906123/197320045-f95ce7d6-a2c6-4b24-b00e-9da72f98cf6c.png">
<img width="504" alt="Screen Shot 2022-10-22 at 12 40 37 AM" src="https://user-images.githubusercontent.com/103906123/197320071-70b1fa65-f4d6-4edc-bcda-6be0585385c9.png">

mike10911 committed:
overflow and build fix

mike10911 committed:
removes code duplication and build fix

mike10911 committed:
import casing build fix

mike10911 committed:
build fix

mike10911 committed:
build fix

mike10911 committed:
response for tablet and tabletLg

mike10911 committed:
cactus getting bigger when toggled

mike10911 commented:
@deanframe @jzhzhang66  
Just added the functionality for the cacti getting bigger when toggled. I have tried and tried but cant get them to be aligned horizontally. Any tips?

jzhzhang66 commented:
Add `  margin-bottom: ${(StyledCactusButtonsProps): string =>
    StyledCactusButtonsProps.isToggled ? '-0.5em' : '0'};`

into your StyledCactusButtons

mike10911 committed:
mobile view

mike10911 commented:
Added Testimonial Section

Fixes #34 

Changelist:

- Cacti getting bigger when toggled by passing in props to the styled component.
- Made testimonial cards slimmer for mobile view.
- Added arrows to navigate left or right through the testimonials.
- When on mobile view only the center testimonial is visible and the cacti navigation buttons are hidden.
- Implemented generics in `getLeftRight()`.
- Implemented suggested fixes from last week.

Tested:

Local Host in multiple browsers and views

Screenshots & Screencasts:

[Home Page.webm](https://user-images.githubusercontent.com/103906123/197645175-9968c3bf-e26a-48d9-b1de-93f75b483339.webm)

mike10911 commented:
Quick Question! @jzhzhang66 @deanframe 
When using generics and defining the type that will be passed in, what style do we want, `<Type>` like in ts documentation, or just `<T>`

jzhzhang66 commented:
@mike10911 <T> is fine! I think that's how Lisa + Sarah are doing it so let's just do it here to so its easier to merge haha

jzhzhang66 commented on main-site/components/testimonials-section/testimonial-card/TestimonialCard.styles.ts:
```diff
@@ -0,0 +1,63 @@
+  width: 22em;
+  height: 28em;
+  @media ${max.tablet} {
+    width: 14em;
```
Can we make this even wider since there's so much text we need to put on it? Like 18em or something and then if possible increase the font size at that screensize. 

You can add a different breakpoint for tabletSm or mobile to configure it more specifically for those screensizes

jzhzhang66 commented on main-site/components/events-calendar-section/EventsCalendarSection.styles.ts:
```diff
@@ -1,205 +1,198 @@
+  }
+  @media ${max.tabletSm} {
+    height: 28em;
+    margin-left: 5%;
```
remove the margin-left: 5% since it's a repeat

mike10911 replied:
In EventsCalenderSection???

mike10911 replied:
oh wait i see

jzhzhang66 commented on main-site/components/events-calendar-section/EventsCalendarSection.styles.ts:
```diff
@@ -1,205 +1,198 @@
+  }
+  @media ${min.tablet} {
+    top: 50em;
+    right: 0;
```
remove right: 0 since its a repeat

jzhzhang66 reviewed (changes requested):
OMG MIKE AWESOME JOB!!!! This is fantastic!!!

A few minor comments but also on Samsung Galaxy S8+, the arrows aren't in the right spot. We should still try to support sizes that are above 300 since there are some phones with those screensizes still. If it's too hard to make it thinner here, you can put both arrows at the bottom like so <      > (ONLY for these super small screensizes though!!! everything else should follow the original design) 
<img width="731" alt="Screen Shot 2022-10-25 at 12 23 32 AM" src="https://user-images.githubusercontent.com/59738880/197682111-513373cd-14e1-4622-8ff4-3fc270bd695d.png">

jzhzhang66 commented:
spencer requested to have his photo changed to be the most recent one he has - this should be in the team preferred headshots folder on the drive

deanframe commented on .husky/pre-commit:
```diff
@@ -3,3 +3,4 @@
 
 npx --no-install lint-staged
 npx --no-install lint-staged
+npx --no-install lint-staged
```
hmm @jzhzhang66 should there be only one of these?

deanframe commented on main-site/components/testimonials-section/TestimonialsSection.styles.ts:
```diff
@@ -0,0 +1,8 @@
+  text-align: center;
+  position: relative;
+  overflow: hidden;
+`;
```
nitpick: let's put a blank line before the export

deanframe reviewed (commented):
agreed with everything judy said!

mike10911 committed:
small mobile phone and style fixes

mike10911 committed:
spacing

mike10911 committed:
merge conflicts

mike10911 committed:
section spacing

mike10911 committed:
spencer year fix

jzhzhang66 reviewed (approved):
LGTM

deanframe reviewed (approved):
lgtm!

mike10911 committed:
merge conflict
